src/overlays/actors/ovl_Bg_Umajump/z_bg_umajump.c

```diff
@@ -27,34 +31,191 @@ const ActorInit Bg_Umajump_InitVars = {
     (ActorFunc)NULL,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_8091A810[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneScale, 1200, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 300, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_8091A810[];
-
-extern UNK_TYPE D_06001220;
-extern UNK_TYPE D_06001558;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Umajump/func_80919F30.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Umajump/func_80919FC8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Umajump/func_8091A044.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Umajump/func_8091A0B8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Umajump/BgUmajump_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Umajump/BgUmajump_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Umajump/BgUmajump_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Umajump/func_8091A5A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Umajump/func_8091A7B0.s")
+void func_80919F30(BgUmajump* this, PlayState* play) {
+    if (this->objectIndex >= 90) {
+        this->objectIndex = 0;
+    }
+
+    this->objectIndex++;
+    this->dyna.actor.shape.yOffset =
+        Math_SinS((this->objectIndex / 90.0f) * 65536.0f) * (20.0f / this->dyna.actor.scale.y);
+}
+
+void BgUmajump_StopCutscene(BgUmajump* this, PlayState* play) {
+    if ((play->csCtx.frames >= 6) && !this->hasSoundPlayed) {
+        this->hasSoundPlayed = true;
+        play_sound(NA_SE_EV_KID_HORSE_NEIGH);
+    }
+
+    if (!play->csCtx.state) {
+        ActorCutscene_Stop(this->dyna.actor.cutscene);
+        this->dyna.actor.update = Actor_Noop;
+    }
+}
+
+void BgUmajump_PlayCutscene(BgUmajump* this, PlayState* play) {
+    if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        gSaveContext.save.weekEventReg[0x59] |= 0x20;
```

Comment:
```suggestion
        gSaveContext.save.weekEventReg[89] |= 0x20;
```
Decimal for weekEventReg indices. There are multiple of these could you please fix the others as well :).